rfcs/select-component.md

```diff
@@ -0,0 +1,191 @@
+# Select Component
+
+### Exports
+
+* `StatefulSelect`
+* `StatefulSelectContainer`
+* `Select`
+* `StyledSelectInput`
+* `StyledOption`
+* `StyledDropDown`
+* `StyledLabel`
+* `StyledHint`
+
+### `Select` and `StatefulSelect` API
+
+* `type: TYPE.SEARCH | TYPE.SELECT`:
+  type of component to be in select or search mode (with lookup input)
+* `selectedOptions: Array<Object>`:
+  Current selected options. Every option object has `id: string` and `label: React$Node`. Label is defaulted to display for selected option, otherwise see `getSelectedOptionLabel` method
+* `placeholder: ?string`:
+  Placeholder text if nothing is selected. Default is `Choose one...`
+* `rows: ?number`:
+  Represents maximum visible length of options, all other will be scrolled. If not defined, all options will be visible.
+* `options: Array<Object>`:
+  All Options in dropdown. Should be provided for Select and Search mode equally. Every option object has `id: string` and `label: React$Node`. Label is defaulted to display for option in dropdown, otherwise see `getOptionLabel` method. Optional `disabled: boolean` for option to be disabled from selection.
+* `label: ?React$Node`:
+  Component or String representing label for Select component. Default is `''`
+* `caption: ?React$Node`:
+  Component or String representing hint message for Select component. Default is `''`
```

Comment:
I think Nadiia's work in #125 will handle this outside of components. Can we remove these?

Reply:
only if it's not a part of Input yet. But so far it is a part of it, which makes it easier for me to use

Reply:
Nadiia will be removing those from input I think, in favor of FormControl. FormControl just wraps existing components, so you don't have to add anything here to support it.

Reply:
removed caption